src/others/components/TimelineSlider.tsx

```diff
@@ -13,15 +21,21 @@ export interface TimelineSliderProps {
 export const TimelineSlider = ({ dates }: TimelineSliderProps) => {
   const [endDate, setEndDate] = useState<moment.Moment>(moment());
   const [startDate, setStartDate] = useState<moment.Moment>(moment());
-  const [selectedDate, setDate] = useState(endDate.toDate());
+  const [selectedDate, setDate] = useState<Date | undefined>(undefined);
   const { toggleFilterItem } = useFilter();
+  const prevDates = usePreviousDates(dates);
 
   useEffect(() => {
+    const endDate = moment(dates[dates.length - 1], inputDateFormat)
     if (dates.length) {
       setStartDate(moment(dates[0], inputDateFormat));
-      setEndDate(moment(dates[dates.length - 1], inputDateFormat));
+      setEndDate(endDate);
+    }
+
+    if (prevDates && prevDates.length !== dates.length) {
```

Comment:
This feels like a bit of a hack. Ideally, I think we should leverage the `isLoading` or `isSuccess` prop from [react-query](https://react-query.tanstack.com/reference/useQuery) when we make the API requests and keep track of that in the filter context. Then, I think we can more reliably track when the API request is complete and also have a better UX when the filters are loading as well.

Reply:
It's definitely a viable solution but feels a bit like overkill. What would you say about this [approach](https://github.com/ukraine-taskforce/requests-dashboard/compare/fix/default-date)? Kind of similar to yours, just using useState instead of useRef. 

Totally agree about isLoading or isSuccess stuff!

Reply:
@wtfua92 Sure, happy to track it via state instead.

I do think we still need to conditionally render the selected date label though, and also not use the current date as the initial state value for `initialDate`. Otherwise, there's a flash in the UI where the selected date shows as today's date, before rendering as the correct date based on the response. 

Let me know what you think.

Reply:
Totally true, let's do it your way for now and we can apply conditional rendering later on unless you wanna do it within the scope of this PR. If no, then it's a LGTM from me